tests/Agent/UnitTests/NewRelic.Agent.Extensions.Tests/Parsing/ConnectionStringParserTests.cs

```diff
@@ -99,13 +116,116 @@ public void ConnectionStringParser_DifferentConnectionStrings_NotMatched()
         public void ConnectionStringParser_UnknownVendor_EmptyResult()
         {
             var connectionString = @"Driver={Microsoft Text Driver (*.txt; *.csv)};Dbq=c:\txtFilesFolder\;Extensions=asc,csv,tab,txt;";
-
-            var connectionInfo = ConnectionInfoParser.FromConnectionString(DatastoreVendor.Other, connectionString, "localhost");
+            ConnectionInfo connectionInfo = null;
+            Assert.DoesNotThrow(() =>
+                {
+                    connectionInfo = ConnectionInfoParser.FromConnectionString(DatastoreVendor.Other, connectionString, "localhost");
+                });
 
             Assert.That(connectionInfo.Host, Is.EqualTo("unknown"));
             Assert.That(connectionInfo.PortPathOrId, Is.EqualTo("unknown"));
             Assert.That(connectionInfo.DatabaseName, Is.EqualTo("unknown"));
             Assert.That(connectionInfo.InstanceName, Is.Null);
         }
+
+        [Test]
+        public void MsSql_MalformedConnectionStrings_DoNotThrow()
```

Comment:
Should we add tests like this for the other datastore libraries that we added the try catch blocks to?